container/go/cmd/pusher/pusher.go

```diff
@@ -56,6 +57,11 @@ func main() {
 		os.Setenv("DOCKER_CONFIG", *clientConfigDir)
 	}
 
+	// Ensure src is a directory, trim basename index.json if not.
+	if filepath.Base(*src) == "index.json" {
+		*src = filepath.Dir(*src)
+	}
+
```

Comment:
This trims `src` to a directory path. I need to do this because ` get_runfile_path(ctx, index.json)` (what we used to find absolute path to an oci-layout in `new_pusher.bzl`) returns a path to `index.json`. I could not think of a good way for this.

Reply:
See comment below.